Data/Aeson/Parser/Internal.hs

```diff
@@ -323,37 +316,20 @@ jstring = A.word8 DOUBLE_QUOTE *> jstring_
 -- | Parse a string without a leading quote.
 jstring_ :: Parser Text
 {-# INLINE jstring_ #-}
-jstring_ = {-# SCC "jstring_" #-} do
-#if MIN_VERSION_ghc_prim(0,3,1)
-  (s, S _ escaped) <- A.runScanner startState go <* A.anyWord8
-  -- We escape only if there are
-  -- non-ascii (over 7bit) characters or backslash present.
-  --
-  -- Note: if/when text will have fast ascii -> text conversion
-  -- (e.g. uses utf8 encoding) we can have further speedup.
-  if isTrue# escaped
-    then case unescapeText s of
-      Right r  -> return r
-      Left err -> fail $ show err
-    else return (TE.decodeUtf8 s)
- where
-    startState              = S 0# 0#
-    go (S skip escaped) (W8# c)
-      | isTrue# skip        = Just (S 0# escaped')
-      | isTrue# (w ==# 34#) = Nothing   -- double quote
-      | otherwise           = Just (S skip' escaped')
-      where
-        w = word2Int# c
-        skip' = w ==# 92# -- backslash
-        escaped' = escaped
-            `orI#` (w `andI#` 0x80# ==# 0x80#) -- c >= 0x80
-            `orI#` skip'
-            `orI#` (w `andI#` 0x1f# ==# w)     -- c < 0x20
```

Comment:
I'm not sure why this check was here. What is special about characters with code less then 0x20 in this context?

Reply:
They're not printable, so they must be escaped.

Reply:
Oh wait, nevermind. We're parsing stuff, so I don't understand either.

Reply:
https://tools.ietf.org/html/rfc7159#section-7, https://tools.ietf.org/html/rfc8259#section-7

E.g. if you try in browser `JSON.parse("\"\u0007\"")`, it will fail. I don't know why current aeson accepts control characters. 

```haskell
decode "\"\7\"" :: Maybe String
Just "\a"
```

We have disabled JSONTestSuite for that https://github.com/bos/aeson/blob/33e0acdd02fde1029cc6c9dd6484cefb00d19a8c/tests/UnitTests.hs#L507-L509, but I really don't remember why. Bryan wrote the tests that way in 2016, and I think this plain wrong, and we should reject the above.

So the `0x1f` check is kind of for that, but in a brain-fart way as it doesn't make parser fail.

Reply:
also https://github.com/bos/aeson/issues/515

Reply:
So the PR doesn't change the behavior. @phadej Thank your for clarification.